addons/payment/views/payment_templates.xml

```diff
@@ -411,7 +452,7 @@
 
     <!-- Test token badge -->
     <template id="test_token_badge" name="Test Token Badge">
-        <span t-if="provider.state == 'test'" class="badge rounded-pill test-bg-warning">
+        <span t-if="token.provider_id.state == 'test'" class="badge rounded-pill text-bg-warning">
```

Comment:
Nice catch for the typo, but why do you change `provider` to `token.provider_id`?